### ISSUE TITLE
[data grid] Keyboard navigation doesn't work when active cell is out of the viewport

### ISSUE THREAD

**Issue by shelimov** [mui/mui-x#5750]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

Whenever I press any of the arrow keys, it applies scroll to the window.
https://codesandbox.io/s/1cm3xe?file=/demo.tsx

https://user-images.githubusercontent.com/4550843/183926380-7b5ce2a4-d6e0-4d5f-bbaa-13314092a9b9.mov

### Expected behavior 🤔

DataGrid should select the next cell and scroll it into view.
The same way it works in [ag-grid](https://www.ag-grid.com/).

https://user-images.githubusercontent.com/4550843/183927567-056100ca-62f4-4b5b-8558-659279faddca.mov

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by DanailH** [comment]
Hi, @shelimov thanks for raising this!
It is easily reproducible. It seems to be related to the focus, I think. @m4theushw I remember we discussed it some time ago but I'm unsure if it is the same problem or not.

**Comment by m4theushw** [comment]
We need to keep the focused cell in the viewport. With virtualization enabled, it's removed as you scroll. I'm working on a POC to fix this and also https://github.com/mui/mui-x/issues/4744. For now, if you're using a small dataset, you can disable the virtualization with `disableVirtualization` and it will work correctly.

**Comment by oliviertassinari** [comment]
I would expect https://github.com/mui/mui-x/issues/4282 and https://github.com/mui/mui-x/issues/4749 to be linked to this problem
